src/syslog.rs

```diff
@@ -0,0 +1,146 @@
+//! Interfaces for controlling system log.
+
+use crate::NixPath;
+use crate::Result;
+use std::ffi::OsStr;
+
+/// Logging options of subsequent [`syslog`] calls can be set by calling [`openlog`].
+///
+/// The parameter `ident` is a string that will be prepended to every message. The `logopt`
+/// argument specifies logging options. The `facility` parameter encodes a default facility to be
+/// assigned to all messages that do not have an explicit facility encoded.
+pub fn openlog<P: NixPath + ?Sized>(
+    ident: &P,
+    logopt: LogFlags,
+    facility: Facility,
+) -> Result<()> {
+    ident.with_nix_path(|ident| unsafe {
+        libc::openlog(ident.as_ptr(), logopt.bits(), facility as libc::c_int);
+    })
+}
+
+/// Writes message to the system message logger.
+///
+/// The message is then written to the system console, log files, logged-in users, or forwarded
+/// to other machines as appropriate.
+pub fn syslog<S: AsRef<OsStr> + ?Sized>(
+    priority: Priority,
+    message: &S,
+) -> Result<()> {
+    let formatter = OsStr::new("%s");
+    let message = OsStr::new(message);
+    formatter.with_nix_path(|formatter| {
+        message.with_nix_path(|message| unsafe {
+            libc::syslog(priority.0, formatter.as_ptr(), message.as_ptr())
+        })
+    })??;
+    Ok(())
+}
+
+/// The priority for a log message.
+#[derive(Debug, Clone, Copy)]
+pub struct Priority(libc::c_int);
+
+impl Priority {
+    /// Create a new priority from a severity level.
+    pub fn from_severity(severity: Severity) -> Self {
+        let priority = severity as libc::c_int;
+        Priority(priority)
+    }
+
+    /// Create a new priority from a facility and severity level.
+    pub fn from(severity: Severity, facility: Facility) -> Self {
+        let priority = (facility as libc::c_int) | (severity as libc::c_int);
+        Priority(priority)
+    }
+}
+
+libc_bitflags! {
+    /// Options for system logging.
+    pub struct LogFlags: libc::c_int {
+        /// Log the process id with each message: useful for identifying instantiations of
+        /// daemons.
+        LOG_PID;
+        /// If syslog() cannot pass the message to syslogd(8) it will attempt to write the
+        /// message to the console ("/dev/console").
+        LOG_CONS;
+        /// Open the connection to syslogd(8) immediately. Normally the open is delayed until
+        /// the first message is logged. Useful for programs that need to manage the order in
+        /// which file descriptors are allocated.
+        LOG_NDELAY;
+        /// Write the message to standard error output as well to the system log.
+        LOG_PERROR;
+    }
+}
+
+libc_enum! {
+    /// Severity levels for log messages.
+    #[repr(i32)]
+    #[non_exhaustive]
+    pub enum Severity {
```

Comment:
The man pages of macOS, FreeBSD, and Linux all use the word "level", what about renaming this type to simply `LogLevel`?

Reply:
Severity is defined as in:

* https://en.wikipedia.org/wiki/Syslog
* https://datatracker.ietf.org/doc/html/rfc3164
* https://datatracker.ietf.org/doc/html/rfc3164

I'd prefer to use this name.

Reply:
Makes sense